app/models/application_reference.rb

```diff
@@ -86,6 +92,16 @@ def find_latest_reference
     .last
   end
 
+  def order_in_application_references
+    return if failed?
+
+    candidate
+      .application_references
+      .not_failed
+      .order(id: :asc)
+      .pluck(:id).index(id) + 1
+  end
```

Comment:
This seems to be ordering them by ID?  The intention was to use "first", "second" etc in the order that the references are received, rather than than order they were originally added to the application. That way, the providers can quickly see if the candidate has received 'enough' references yet.